includes/class-llms-block-templates.php

```diff
@@ -0,0 +1,564 @@
+	 */
+	public function add_llms_block_templates( $query_result, $query, $template_type = 'wp_template' ) {
+
+		// Bail it's not a block theme, or is being retrieved a non wp_template type requested.
+		if ( ! function_exists( 'wp_is_block_theme' ) || ! wp_is_block_theme() || 'wp_template' !== $template_type ) {
+			return $query_result;
+		}
+
+		$post_type = $query['post_type'] ?? '';
+		$slugs     = $query['slug__in'] ?? array();
+
+		// Retrieve templates.
+		$templates = $this->block_templates( $slugs, $post_type );
+
+		/**
+		 * Remove theme override templates who have a customization in the db from $query_result:
+		 * those template blocks will be already retrieved by our LLMS_Block_Templates::block_templates_from_db().
+		 */
+		$query_result = array_values(
+			array_filter(
+				$query_result,
+				function( $template ) use ( $templates ) {
+					$slugs = wp_list_pluck( $templates, 'slug' );
+					return ( ! in_array( $template->slug, $slugs, true ) );
+				}
+			)
+		);
+
+		return array_merge( $query_result, $templates );
+
+	}
+
+	/**
+	 * Returns an associative array of template titles.
+	 *
+	 * Keys are unprefixed template slugs.
+	 * Values are template titles in a human readable form.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	private function template_titles() {
+
+		$template_titles = $this->template_titles ?? array(
+			'archive-course'             => esc_html__( 'Course Catalog', 'lifterlms' ),
+			'archive-llms_membership'    => esc_html__( 'Membership Catalog', 'lifterlms' ),
+			'single-certificate'         => esc_html__( 'Single Certificate', 'lifterlms' ),
+			'single-no-access'           => esc_html__( 'Single Requiring Membership', 'lifterlms' ),
+			'taxonomy-course_cat'        => esc_html__( 'Taxonomy Course Category', 'lifterlms' ),
+			'taxonomy-course_difficulty' => esc_html__( 'Taxonomy Course Difficulty', 'lifterlms' ),
+			'taxonomy-course_tag'        => esc_html__( 'Taxonomy Course Tag', 'lifterlms' ),
+			'taxonomy-course_track'      => esc_html__( 'Taxonomy Course Track', 'lifterlms' ),
+			'taxonomy-memberhsip_cat'    => esc_html__( 'Taxonomy Membership Category', 'lifterlms' ),
+			'taxonomy-memberhsip_tag'    => esc_html__( 'Taxonomy Membership Tag', 'lifterlms' ),
+		);
+
+		$this->template_titles = $this->template_titles ?? $template_titles;
+
+		/**
+		 * Filters the block template titles.
+		 *
+		 * @since [version]
+		 *
+		 * @param array $template_titles  {
+		 *     Associative array of template titles.
+		 *
+		 *     @type string $slug  The template slug (unprefixed).
+		 *     @type string $title The template readable titles.
+		 * }
+		 */
+		return apply_filters( 'lifterlms_block_templates_titles', $this->template_titles );
+
+	}
+
+	/**
+	 * Block Templates admin js strings.
+	 *
+	 * @since [version]
+	 *
+	 * @param string[] $strings Localization admin strings.
+	 * @return string[]
+	 */
+	public function add_block_templates_admin_strings_js( $strings ) {
```

Comment:
Well my idea is that this was more easy to extend, in an add-on.
E.g. I'm going to add a block template for the groups plugin, and I wanted to only add the template html + a php file where I add all the filters needed to load the template, and its js title, correctly.
But that's fine I'll add some inline script in groups to do the same (and a filter hook in https://github.com/gocodebox/lifterlms-blocks/pull/147/files#diff-c12382294566f22f66d320592d72c6293bae6b752c5b4eb0296d7990685d81c5R4). As I don't feel to add a js file only for that in groups.

Reply:
@eri-trabiccolo I'm find with localizing with PHP instead of JS for lighter-weight extensibility in this case. However, I don't see any reason to use `LLMS.l10n` in order to do so.

Instead, keep this but turn it into a JS localization object and then in JS we can read the map to output the object instead of loading them into `LLMS.l10n` and reading them that way. I think that will work, you?

Reply:
so I'm going to localize our blocks script in a more formal way?!

Reply:
@eri-trabiccolo I don't think it needs to be anything more formal than outputting an object of translation data, mapping the template ID/name to the translated title, and outputting it as a JS variable using [wp_localize_script()](https://developer.wordpress.org/reference/functions/wp_localize_script/). Add a filter on the object before it's passed into the function. Then add-ons can add their own data to it via PHP.

In the legacy block JS, read the map I hard-coded into the `getLabel()` method from the localized JS object instead of it being hard coded.

Reply:
oh, with formal I meant use `wp_localize_script()` yeah :D, rather than build the object as we do in other parts of the codebase. <3

Reply:
php-block templates localized